agave_app/CameraWidget.cpp

```diff
@@ -4,28 +4,154 @@
+  m_MainLayout.addRow("Noise Reduction", &m_NoiseReduction);
+
+  connect(&m_NoiseReduction, &QCheckBox::stateChanged, this, &QCameraWidget::OnNoiseReduction);
+
+  m_ApertureSizeSlider.setStatusTip(tr("Set camera aperture size"));
+  m_ApertureSizeSlider.setToolTip(tr("Set camera aperture size"));
+  m_ApertureSizeSlider.setRange(0.0, 0.1);
+  m_ApertureSizeSlider.setSuffix(" mm");
+  m_ApertureSizeSlider.setDecimals(2);
+  m_ApertureSizeSlider.setValue(0.0);
+  m_ApertureSizeSlider.setSingleStep(0.01);
+  m_MainLayout.addRow("Aperture Size", &m_ApertureSizeSlider);
+
+  connect(&m_ApertureSizeSlider, &QNumericSlider::valueChanged, this, &QCameraWidget::SetAperture);
+
+  m_FieldOfViewSlider.setStatusTip(tr("Set camera field of view angle"));
+  m_FieldOfViewSlider.setToolTip(tr("Set camera field of view angle"));
+  m_FieldOfViewSlider.setRange(10.0, 150.0);
+  m_FieldOfViewSlider.setDecimals(2);
+  m_FieldOfViewSlider.setValue(cam->GetProjection().GetFieldOfView());
+  m_FieldOfViewSlider.setSuffix(" deg.");
+  m_MainLayout.addRow("Field of view", &m_FieldOfViewSlider);
+
+  connect(&m_FieldOfViewSlider, &QNumericSlider::valueChanged, this, &QCameraWidget::SetFieldOfView);
+
+  // Focal distance
+  m_FocalDistanceSlider.setStatusTip(tr("Set focal distance"));
+  m_FocalDistanceSlider.setToolTip(tr("Set focal distance"));
+  m_FocalDistanceSlider.setRange(0.0, 15.0);
+  m_FocalDistanceSlider.setDecimals(2);
+  m_FocalDistanceSlider.setValue(0.0);
+  m_FocalDistanceSlider.setSuffix(" m");
+
+  m_MainLayout.addRow("Focal distance", &m_FocalDistanceSlider);
+
+  connect(&m_FocalDistanceSlider, &QNumericSlider::valueChanged, this, &QCameraWidget::SetFocalDistance);
+
+  QObject::connect(&cam->GetFilm(), SIGNAL(Changed(const QFilm&)), this, SLOT(OnFilmChanged()));
+  QObject::connect(&cam->GetAperture(), SIGNAL(Changed(const QAperture&)), this, SLOT(OnApertureChanged()));
+  QObject::connect(&cam->GetProjection(), SIGNAL(Changed(const QProjection&)), this, SLOT(OnProjectionChanged()));
+  QObject::connect(&cam->GetFocus(), SIGNAL(Changed(const QFocus&)), this, SLOT(OnFocusChanged()));
+}
+
+void
+QCameraWidget::OnFilmChanged()
+{
+  m_ExposureSlider.setValue(m_qcamera->GetFilm().GetExposure(), true);
+  m_ExposureIterationsSpinner.blockSignals(true);
+  m_ExposureIterationsSpinner.setCurrentIndex(
+    m_ExposureIterationsSpinner.findData(m_qcamera->GetFilm().GetExposureIterations()));
+  m_ExposureIterationsSpinner.blockSignals(false);
+  m_NoiseReduction.blockSignals(true);
+  m_NoiseReduction.setCheckState(m_renderSettings->m_DenoiseParams.m_Enabled ? Qt::CheckState::Checked
+                                                                             : Qt::CheckState::Unchecked);
+  m_NoiseReduction.blockSignals(false);
+  emit m_qcamera->Changed();
+}
+void
+QCameraWidget::OnApertureChanged()
+{
+  m_ApertureSizeSlider.setValue(m_qcamera->GetAperture().GetSize(), true);
+  emit m_qcamera->Changed();

```

Comment:
Do these need to be configured to block signals while being changed too?

Reply:
the second argument to setValue is a boolean blockSignals parameter.  The emit m_qCamera0->Changed() is just a signal to say that the camera was updated but not via the UI.  This codepath is for receiving changes when a saved session is loaded from json, and we need to make sure the UI is updated.  So it allows two-way communication between the UI and QObjects holding state.